api/v1alpha1/spannerautoscaler_conversion_test.go

```diff
@@ -0,0 +1,331 @@
+package v1alpha1
+
+import (
+	"github.com/mercari/spanner-autoscaler/api/v1beta1"
+	"github.com/mercari/spanner-autoscaler/utils/pointer"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+)
+
+func TestSpannerAutoscaler_ConvertTo(t *testing.T) {
+	t.Parallel()
+	timestamp := time.Now()
+
+	tests := []struct {
+		title    string
+		src      *SpannerAutoscaler
+		expected *v1beta1.SpannerAutoscaler
+	}{
+		{
+			title: "target instance",
+			src: &SpannerAutoscaler{
+				Spec: SpannerAutoscalerSpec{
+					ScaleTargetRef: ScaleTargetRef{
+						ProjectID:  pointer.String("src-project-id"),
+						InstanceID: pointer.String("src-instance-id"),
+					},
+					TargetCPUUtilization: TargetCPUUtilization{
+						HighPriority: pointer.Int32(50),
+					},
+				},
+			},
+			expected: &v1beta1.SpannerAutoscaler{
+				Spec: v1beta1.SpannerAutoscalerSpec{
+					TargetInstance: v1beta1.TargetInstance{
+						ProjectID:  "src-project-id",
+						InstanceID: "src-instance-id",
+					},
+					ScaleConfig: v1beta1.ScaleConfig{
+						TargetCPUUtilization: v1beta1.TargetCPUUtilization{
+							HighPriority: 50,
+						},
+					},
+				},
+			},
+		},
+		{
+			title: "authentication AuthTypeImpersonation",
+			src: &SpannerAutoscaler{
+				Spec: SpannerAutoscalerSpec{
+					ScaleTargetRef: ScaleTargetRef{
+						ProjectID:  pointer.String("src-project-id"),
+						InstanceID: pointer.String("src-instance-id"),
+					},
```

Comment:
DRY the code wherever possible. This (as well as some other sections) are duplicated and repeatedly typed over and over in this test file. This should be avoided.